src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/RegularExpressionAttribute.cs

```diff
@@ -21,9 +21,16 @@ public RegularExpressionAttribute(string pattern)
             : base(() => SR.RegexAttribute_ValidationError)
         {
             Pattern = pattern;
+            MatchTimeoutInMilliseconds = 2000;
         }
 
         /// <summary>
+        ///     Gets or sets the timeout to use when matching the regular expression pattern (in milliseconds)
+        ///     (-1 means never timeout).
+        /// </summary>
+        public int MatchTimeoutInMilliseconds { get; set; }
```

Comment:
Yes just in the various csproj files should take care of it (at least the ref and src csproj files).